plugins/maxruntime/maxruntime.go

```diff
@@ -0,0 +1,88 @@
+package maxruntime
+
+import (
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	schematypes "github.com/taskcluster/go-schematypes"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/plugins"
+)
+
+type pluginProvider struct {
+	plugins.PluginProviderBase
+}
+
+type plugin struct {
+	plugins.PluginBase
+}
+
+type taskPlugin struct {
+	plugins.TaskPluginBase
+	maxRunTime int
+	done       chan bool
+	log        *logrus.Entry
+}
+
+type payloadType struct {
+	MaxRunTime int `json:"maxRunTime"`
+}
+
+var payloadSchema = schematypes.Object{
+	Properties: schematypes.Properties{
+		"maxRunTime": schematypes.Integer{
+			MetaData: schematypes.MetaData{
+				Title:       "Task execution timeout (ms)",
```

Comment:
@jonasfj Good question! Although ms has the advantage that you can specify sub-second intervals, it comes with the disadvantage that numbers just get bigger, e.g. one hour is 3600000 which at a glance is not easy to recognise as an hour (you have to carefully count the 0's). Since the time overhead of a task is at least a second, I think it isn't optimal if we plan for tasks that run for less than a second (i.e. we wouldn't optimise chunking so that tasks last less than a second) so it feels unnecessary to support this level of accuracy. I think `maxRunTime` should really be considered a safety net, so if your task runs for a crazy amount of time, it can be aborted, rather than it being used as a mechanism to intentionally fail tasks that don't execute in a precise amount of time - and as a safety mechanism, I would say seconds would be the absolute smallest unit we'd want to consider - I would also not be put off using minutes, although I maybe seconds is a safer middle ground, and is consistent with current worker implementations (at least, that is what generic worker uses).